lib/internal/streams/pipeline.js

```diff
@@ -102,25 +106,52 @@ async function* fromReadable(val) {
 }
 
 async function pump(iterable, writable, finish) {
-  if (!EE) {
-    EE = require('events');
-  }
   let error;
+  let callback = noop;
+  const resume = (err) => {
+    if (!error && err) {
+      error = err;
+    }
```

Comment:
```suggestion
    error ||= err;
```
nit: This way, we could avoid the `if`.